File: app/src/main/java/cn/nicolite/huthelper/view/activity/SplashActivity.java
package cn.nicolite.huthelper.view.activity;

import android.text.TextUtils;

import java.util.List;
import java.util.Timer;
import java.util.TimerTask;

import cn.nicolite.huthelper.R;
import cn.nicolite.huthelper.base.BaseActivity;
import cn.nicolite.huthelper.model.bean.Configure;
import cn.nicolite.huthelper.utils.ListUtils;

/**
 * 闪屏页
 * Created by nicolite on 17-10-17.
 */

public class SplashActivity extends BaseActivity {
    private Timer timer = null;

    @Override
    protected int setLayoutId() {
        return R.layout.activity_splash;
    }

    @Override
    protected void doBusiness() {
        timer = timer == null ? new Timer() : timer;
        timer.schedule(new TimerTask() {
            @Override
            public void run() {
                if (isLogin()) {
                    startActivity(MainActivity.class);
                } else {
                    startActivity(LoginActivity.class);
                }
            }
        }, 680);
    }

    public boolean isLogin() {
        //对1.3.8以前的版本做兼容，需要重新登录, 版本迭代完成后可删除
        List<Configure> configureList = getConfigureList();
        if (!ListUtils.isEmpty(configureList)) {
            if (TextUtils.isEmpty(configureList.get(0).getStudentKH())) {
                return false;
            }
        }
        return userId != null && !userId.equals("*") && !ListUtils.isEmpty(getConfigureList());
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (timer != null) {
            finish();
        }
    }

    @Override
    protected void onDestroy() {
        timer = null;
        super.onDestroy();
    }
}

File: app/src/main/java/cn/nicolite/huthelper/view/activity/SplashActivity.kt
package cn.nicolite.huthelper.view.activity

import android.os.Bundle
import android.text.TextUtils
import cn.nicolite.huthelper.R
import cn.nicolite.huthelper.kBase.BaseActivity
import cn.nicolite.huthelper.utils.ListUtils

/**
 * 闪屏页
 * Created by nicolite on 17-10-17.
 */

class SplashActivity : BaseActivity() {

    private val bgs = intArrayOf(R.drawable.start_1, R.drawable.start_2, R.drawable.start_3, R.drawable.start_4, R.drawable.start_5)

    fun isLogin(): Boolean {
        //对1.3.8以前的版本做兼容，需要重新登录, 版本迭代完成后可删除
        if (!ListUtils.isEmpty(configureList)) {
            if (TextUtils.isEmpty(configureList[0].studentKH)) {
                return false
            }
        }
        return loginUserId != "*" && !ListUtils.isEmpty(configureList)
    }

    override fun initConfig(savedInstanceState: Bundle?) {
        setImmersiveStatusBar()
    }


    override fun setLayoutId(): Int {
        return R.layout.activity_splash
    }

    override fun doBusiness() {
        startMainActivity()
    }

    private fun startMainActivity() {
        if (isLogin()) {
            startActivity(MainActivity::class.java)
        } else {
            startActivity(LoginActivity::class.java)
        }
        finish()
    }
}
